Extend the client to iterate over historical weather forecast data.

david-natingga-frequenz opened the pull request:
Extend the client to iterate over historical weather forecast data.

Add examples to demonstrate:
- How to stream live data with the client based on [Christoph's example](https://github.com/frequenz-floss/frequenz-api-weather/issues/83#issuecomment-1968911497).
- How to iterate over the historical data with the client.

Example output from the second example:
```
Timestamp: seconds: 1702940400

feature: FORECAST_FEATURE_V_WIND_COMPONENT_100_METRE
value: 5.11610031

feature: FORECAST_FEATURE_U_WIND_COMPONENT_100_METRE
value: 12.4870567

Timestamp: seconds: 1702944000

feature: FORECAST_FEATURE_V_WIND_COMPONENT_100_METRE
value: 5.08150578

feature: FORECAST_FEATURE_U_WIND_COMPONENT_100_METRE
value: 12.0456848
```

david-natingga-frequenz committed:
Extend the client to iterate over historical weather forecast data.

Signed-off-by: David Natingga <david.natingga@frequenz.com>

TalweSingh reviewed (approved):
LGTM! Just a minor nitpick. You could split the commit into two. One for the Iterator and one for the examples.

david-natingga-frequenz committed:
Add examples for the client.

Examples demonstrate:
- How to stream live data with the client.
- How to iterate over the historical data with the client.

Signed-off-by: David Natingga <david.natingga@frequenz.com>
